internal/gcsx/random_reader.go

```diff
@@ -350,9 +350,12 @@ func (rr *randomReader) ReadAt(
 	// is a 15-20x improvement in throughput: 150-200 MB/s instead of 10 MB/s.
 	if rr.reader != nil && rr.start < offset && offset-rr.start < maxReadSize {
 		bytesToSkip := offset - rr.start
-		p := make([]byte, bytesToSkip)
-		n, _ := io.ReadFull(rr.reader, p)
-		rr.start += int64(n)
+		discardedBytes, copyError := io.CopyN(io.Discard, rr.reader, int64(bytesToSkip))
+		// io.EOF is expected when reader is shorter than required offset to read.
```

Comment:
Not blocking this PR but I could see this warning 
```
The double modal "required offset" is nonstandard (only accepted in certain dialects). Consider to be offset.
```
required offset --> offset.